feat: mostly playable game, with timers, scoring, and appropriate words!

jacoblb64 opened the pull request:
- filter out words with repeated characters
- make the game strategy longer (still errors at the end)
- update html title
- scoring! simple incrementing for now
- explainer text that goes away on first valid keydown
- added a timer to the top right corner
- when timer expires, if you're still on the same challenge, go to game over
- updated game over screen
- known bug: restarting from gameover just continues, doesn't reset word and score
- dynamic scoring! you get more points the faster you finish the challenge

Here's a video in action:
https://github.com/user-attachments/assets/f3e9c52a-b360-4189-b8b2-f1e442d4790f

jacoblb64 committed:
feat: Make everything more playable

- Filter out words with repeated characters
- Make the game strategy longer (still errors at the end)
- Update html title
- Scoring! Simple incrementing for now
- Explainer text that goes away on first valid keydown

jacoblb64 committed:
feat: working timer and gameplay loop

- added a timer to the top right corner
- when timer expires, if you're still on the same challenge, go to game over
- updated game over screen
- known bug: restarting from gameover just continues, doesn't reset word and score

jacoblb64 committed:
feat: dynamic scoring

jacoblb64 committed:
fix: boot issue

evhabs commented on src/game/libs/GameEngine.ts:
```diff
@@ -21,19 +21,22 @@ export function getStrategyA(): Challenge[] {
+        for (let i = 0; i <= 10; i++) {
+            challenges.push({ characterLength: j, time: 1 });
+        }
+    }
```
😂😂😂 At a max of 10 characters I'd assume if someone beats this they def have someone else helping them on the keyboard HAHAHA

castavridis replied:
hahaha

jacoblb64 replied:
The game still errors if you get to the end of this array, so this felt like a good way to make sure it never happens 😂

evhabs reviewed (approved):
Looks good to me!

castavridis reviewed (approved):
Quick review. Things function as expected and the improvements are make it feel like a bonafide game. UI improvement, I didn't notice the timer countdown while I was playing until I started this review!

jacoblb64 committed:
feat: more visible timer and slighly easier timings

- moved the timer farther in (75 instead of 32px)
- made the timer bigger
- started from 6 seconds
- "expert mode" is now 3 seconds instead of 1

jacoblb64 commented:
> Quick review. Things function as expected and the improvements are make it feel like a bonafide game. UI improvement, I didn't notice the timer countdown while I was playing until I started this review!

I tested this on my laptop instead of my monitor and realized it was getting cut off in certain circumstances -- I moved it farther in which should hopefully help!
